### PR TITLE
Config value objects v2

### PR DESCRIPTION
**Pull request by webrouse**
Changes:
- Added new configuration value objects + tests.
- Modified `TableNodesDecorator`, each node has own "factory" method, as recommended by Thomas: https://github.com/keboola/db-extractor-config/pull/7#discussion_r426472439

### CODE REVIEW

**Comment by webrouse** [comment]
@ondrajodas @tomasfejfar 

Vyhodil som z value objectov validaciu, ako sme sa dohodli. 

Defaultne hodnoty som tiez presunul do `TableNodesDecorator`.

Napisal som "klasicke" testy, kedze teraz sa v tych value objectoch riesi ovela menej veci, takze toho nie je az tak vela.

File: src/Configuration/ValueObject/ExportConfig.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+        return $this->table;
+    }
+
+    public function isIncrementalFetching(): bool
+    {
+        return $this->incrementalFetchingConfig !== null;
+    }
+
+    public function hasIncrementalLimit(): bool
```
Comment (ondrajodas) [inline src/Configuration/ValueObject/ExportConfig.php:84]: spíš `hasIncrementalFetchingLimit`

Comment (webrouse) [reply src/Configuration/ValueObject/ExportConfig.php:84]: vdaka. to som prehliadol.

Comment (webrouse) [reply src/Configuration/ValueObject/ExportConfig.php]: fixnute v https://github.com/keboola/db-extractor-config/pull/8/commits/9299716d3a9aaaafab94e261df0cb9ee785e1c50

**Pull request merged**